### ISSUE TITLE
LSP: autocomplete and validate fenced code block language labels via GitHub Linguist `languages.yml`

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#377]
## Summary

Please add LSP support for autocompletion and validation of fenced code block language labels in Markdown files, using GitHub Linguist `languages.yml` as the source of truth.

This should align with `code-block-tools` language normalization (`normalize-language = "linguist"`) introduced in #331, so editors can catch typos early and offer correct language labels while typing.

## Desired LSP behavior

When the cursor is in a fenced code block info string (for example after ``` or ~~~):

- Offer completion items for all language labels supported by GitHub Linguist.
- Treat known Linguist aliases as valid (not an error).
- If `code-block-tools.normalize-language = "linguist"`, prefer suggesting canonical Linguist language names in completion.
- If `code-block-tools.language-aliases` maps an alias to a canonical language, treat the alias as valid and prefer the canonical target in suggestions.
- If the label is not recognized (not in Linguist names or aliases, and not covered by `language-aliases`), publish a diagnostic on the info string.

## Acceptance criteria

- Completion works for fenced code blocks using backticks and tildes.
- Diagnostics appear only for unknown labels, and do not trigger for unlabeled fences.
- The language list is derived from GitHub Linguist `languages.yml` and stays in sync when it changes (same approach as the schema work in #331).

## References

- `code-block-tools` docs: https://github.com/rvben/rumdl/blob/main/docs/code-block-tools.md
- GitHub Linguist languages list: https://github.com/github-linguist/linguist/blob/main/lib/linguist/languages.yml
- Related issue: https://github.com/rvben/rumdl/issues/331

**Comment by rvben** [comment]
Implemented in v0.1.13. The LSP now provides code block language completion:

- Type ` ``` ` and trigger completion (Ctrl+Space) for language suggestions
- Includes 100+ languages from GitHub Linguist
- Recognizes language aliases (e.g., `py` → `python`)

See [docs/lsp.md](https://github.com/rvben/rumdl/blob/main/docs/lsp.md) for details.
